projects/pgai/pgai/cli.py

```diff
@@ -243,3 +245,764 @@ def install(db_url: str, strict: bool) -> None:
+    help="The name of the semantic catalog to generate embeddings for.",
+)
+@click.option(
+    "-e",
+    "--embed-config",
+    type=click.STRING,
+    default=None,
+    help="The name of the embedding configuration to generate embeddings for. (If None, do all)",  # noqa: E501
+)
+@click.option(
+    "--base-url",
+    type=click.STRING,
+    default=None,
+    help="The base_url for the embedding provider",
+)
+@click.option(
+    "--api-key-name",
+    type=click.STRING,
+    default=None,
+    help="The name of the environment variable containing the API key for the embedding provider",  # noqa: E501
+)
+@click.option(
+    "-q",
+    "--quiet",
+    is_flag=True,
+    help="Do not print log messages.",
+)
+@click.option(
+    "-l",
+    "--log-file",
+    type=click.Path(dir_okay=False, writable=True, path_type=Path),
+    default=None,
+    help="The path to a file to write log messages to.",
+)
+@click.option(
+    "--log-level",
+    type=click.Choice(
+        ["DEBUG", "INFO", "WARN", "ERROR", "FATAL", "CRITICAL"], case_sensitive=False
+    ),
+    default="INFO",
+)
+def create(
+    db_url: str,
+    provider: str,
+    model: str,
+    vector_dimensions: int,
+    catalog_name: str | None = None,
+    embed_config: str | None = None,
+    base_url: str | None = None,
+    api_key_name: str | None = None,
+    quiet: bool = False,
+    log_file: Path | None = None,
+    log_level: str | None = "INFO",
+):
+    import logging
+
+    log_handlers: list[logging.Handler] = []
+    if log_file:
+        log_handlers.append(logging.FileHandler(log_file.expanduser().resolve()))
+    if not quiet:
+        from rich.console import Console
+        from rich.logging import RichHandler
+
+        log_handlers.append(
+            RichHandler(console=Console(stderr=True), rich_tracebacks=True)
+        )
+
+    if len(log_handlers) > 0:
+        logging.basicConfig(
+            level=get_log_level(log_level or "INFO"),
+            format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+            handlers=log_handlers,
+        )
+
+    from pgai.semantic_catalog.vectorizer import (
+        OllamaConfig,
+        OpenAIConfig,
+        SentenceTransformersConfig,
+    )
+
+    match provider.lower():
+        case "sentence_transformers":
+            config = SentenceTransformersConfig.create(
+                model=str(model),
+                dimensions=int(vector_dimensions),
+            )
+        case "openai":
+            config = OpenAIConfig.create(
+                model=str(model),
+                dimensions=int(vector_dimensions),
+                base_url=base_url,
+                api_key_name=api_key_name,
+            )
+        case "ollama":
+            config = OllamaConfig.create(
+                model=str(model),
+                dimensions=int(vector_dimensions),
+                base_url=base_url,
+            )
+        case _:
+            raise RuntimeError(f"unrecognized provider {provider}")
```

Comment:
I think that we should move this into a utility function if not just have it be part of `pgai.semantic_catalog.create`. Otherwise the API for creating a semantic catalog programmatically is not really usable imo as it's super leaky for creating the config.